motionplan/motionPlanner.go

```diff
@@ -175,6 +185,43 @@ func Replan(ctx context.Context, request *PlanRequest, currentPlan Plan, replanC
 		return nil, errors.New("solver frame has no degrees of freedom, cannot perform inverse kinematics")
 	}
 
+	// Here we determine if we already are at the goal.
+	// If our motion profile is position_only then, we only check against our current & desired position.
+	// Conversely if our motion profile is anything else, then we also need to check again our
+	// current & desired orientation.
+	// If we are already at the goal we return an empty plan.
+	profile, ok := request.Options["motion_profile"].(string)
```

Comment:
I'm not sure why we would be changing anything in the motionplan package.

Reply:
https://github.com/viamrobotics/rdk/pull/4269#discussion_r1733381424

Reply:
Peter suggested that this check should be handled by the motion planner, not by the motion service.